llm-translator.html

```diff
@@ -0,0 +1,191 @@
+        const subtitleDiv = document.getElementById("subtitles-" + paxId);
+        const p = document.createElement("p");
+        p.innerHTML = message;
+        subtitleDiv.appendChild(p);
+        setTimeout(() => {
+          p.remove();
+        }, 5000);
+      }
+
+      async function main() {
+        const urlParams = new URLSearchParams(window.location.search);
+        const domain = urlParams.get("domain") || "chad-hq";
+        const room = urlParams.get("room") || "sfu";
+
+        const ROOM_URL = `https://${domain}.daily.co/${room}`;
+        //const ROOM_URL="https://preview-9024.staging.daily.co/raw-tracks?domain=chad-hq&bypassRegionDetection=true"
+        window.call = DailyIframe.createCallObject({
+          url: ROOM_URL,
+          //startAudioOff: true,
+          //startVideoOff: true,
+          subscribeToTracksAutomatically: false,
+          audioSource: false,
+          videoSource: false,
+          dailyConfig: {},
+        });
+
+        [
+          "participant-updated",
+          "participant-left",
+          "error",
+          "camera-error",
+          "started-camera",
+          "active-speaker-change",
+          "recording-started",
+          "recording-stopped",
+          "recording-error",
+          "track-started",
+          "track-stopped",
+        ].forEach((e) => {
+          call.on(e, (ev) => {
+            console.log(`🔥 ${e}:`, ev);
+          });
+        });
+
+        call.on("track-started", startTrack);
+        call.on("track-stopped", stopTrack);
+
+        call.on("participant-joined", (ev) => {
+          console.log("🤠 Participant joined: ", ev);
+          const p = ev.participant;
+          // is this a person or a translator bot?
+          if (isTranslator(p)) {
+            // This is a translator. Update the list of languages
+            // to include this translator's language
+            console.log("this is a translator");
+            const lang = translatorLanguage(p);
+            const newOpt = document.createElement("option");
+            newOpt.value = lang;
+            newOpt.text = cap(lang);
+            languageSelect.appendChild(newOpt);
+          } else {
+            // This is a human. Subscribe to their video, and their
+            // audio if the current language is English
+            console.log("this is a human");
+            console.log(languageSelect.value);
+            const audioSub = languageSelect.value === "english";
+            call.updateParticipant(p["session_id"], {
+              setSubscribedTracks: {
+                video: true,
+                audio: audioSub,
+              },
+            });
+          }
+        });
+
+        call.on("app-message", (e) => {
+          console.log("Got app message: ", e);
+          const lang = languageSelect.value;
+          if (e.fromId === "transcription" && lang === "english") {
+            console.log("got it for english: ", e.data.text);
```

Comment:
If we want to keep these, let's make the text a little more descriptive (i.e., got what for English?)